api/v1alpha1/progressiverollout_types.go

```diff
@@ -100,6 +102,34 @@ func (in *ProgressiveRollout) Owns(owners []metav1.OwnerReference) bool {
 	return false
 }
 
+// SetStageStatus sets the corresponding StageStatus in stageStatus to newStatus
+// - If a stage doesn't exist, it will be added to StageStatus slice
+// - If a stage already exists it will be updated
+func (in *ProgressiveRollout) SetStageStatus(newStatus StageStatus, updateTime *metav1.Time) {
```

Comment:
we briefly mentioned it earlier, but is there a reason behind the pointer semantics of  `metav1.Time`?

Reply:
On this, we kept the time as a pointer in the struct for the reasons Matteo mentioned, but changed it in this method's signature - I can change it back if we go with pointer semantics here anyway.